src/watchers/repeat.js

```diff
@@ -1,126 +1,137 @@
+  var elements = Silica.querySorted(this, "[data-repeat]");
   for (let i = 0, length = elements.length; i < length; ++i) {
-    raw = elements[i]
-    repeat = raw.dataset['repeat'].split(/\s+in\s+/)
-    list = repeat[1]
-    model = repeat[0]
-    ctx = Silica.getContext(raw)
+    raw = elements[i];
+    repeat = raw.dataset["repeat"].split(/\s+in\s+/);
+    list = repeat[1];
+    model = repeat[0];
+    ctx = Silica.getContext(raw);
+    let possiblyNested = !raw.dataset["repeatNotNested"];
 
     // Check if we are calling a function with a param
-    if (typeof (_ref = list.match(/((?:\w|\.)+)(?:\(([\w\.]+)\))*/))[2] !== 'undefined') {
-      let param = _ref[2]
-      param = Silica.getValue(raw.parentNode, param)
-
-      newList = Silica.getValue(raw, _ref[1], null, param)
+    if (
+      typeof (_ref = list.match(/((?:\w|\.)+)(?:\(([\w\.]+)\))*/))[2] !==
+      "undefined"
+    ) {
+      let param = _ref[2];
+      param = Silica.getValue(raw.parentNode, param);
+
+      newList = Silica.getValue(raw, _ref[1], null, param);
     } else {
-      newList = Silica.getValue(raw, list)
-    }
-
-    newListHash = Silica.hasher(JSON.stringify(newList, function (key, value) {
-      // Keys starting with an underscore (char code 95) will be ignored
-      if (key.constructor === String && (key === '__elm' || key === '$ctrl' || key.charCodeAt(0) === 95)) {
-        return undefined
-      }
-      return value
-    }))
-
-    let existing = raw.childNodes
-    oldList = raw._rt_repeat_list
-    changed = oldList && newList ? oldList !== newListHash : true
-
-    if (!changed) {
-      continue
+      newList = Silica.getValue(raw, list);
     }
 
-    if (newList) {
-      raw._rt_repeat_list = newListHash
-    } else {
-      raw._rt_repeat_list = null
-    }
+    let existing = raw.childNodes;
 
     if (!newList) {
       while (raw.childNodes.length > 0) {
-        Silica.removeFromDOM(raw.childNodes[0])
+        Silica.removeFromDOM(raw.childNodes[0]);
       }
-      continue
+      continue;
     }
 
     // Add support for iterating X times
     if (newList.constructor === Number) {
-      newList = new Array(newList)
+      newList = new Array(newList);
     }
 
     // Get the template
-    template = Silica._repeat_templates[Hax.getDatasetProperty(raw, '_rt_repeat_template')]
+    template = Silica._repeat_templates[raw.dataset["siO2TemplateId"]];
     if (newList.constructor === Object) {
-      let keys = Object.keys(newList)
-      let obj = newList
-      newList = []
+      let keys = Object.keys(newList);
+      let obj = newList;
+      newList = [];
       for (let j = 0, len = keys.length, key = keys[j]; j < len; j++) {
-        key = keys[j]
-        newList[j] = { key: key, value: obj[key] }
+        key = keys[j];
+        newList[j] = { key: key, value: obj[key] };
       }
     }
 
-    let countDiff = raw.childElementCount - newList.length
-    let node
+    let countDiff = raw.childElementCount - newList.length;
```

Comment:
guys I think there might be a bug here. (or maybe is due to how I'm using Silica -still brand new 😝 )

Given the following context:
A data repeat on a `<select>` element with initial options: `A, B, C`;
if the model changes to 'D, E' we are keeping C;

As seen here:
https://www.loom.com/share/7ddbd084a67c4b2ba661a3f7ef6d6682

Reply:
Thanks - for the report. Solved by bfb9f06